### PR TITLE
[machine,node] write commitments for install, update, uninstall

### PR DESCRIPTION
**Pull request by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll**

### CODE REVIEW

File: packages/node/src/store.ts [hunk 1]
```diff
@@ -231,6 +231,7 @@ export class Store {
             ["string", "string", "string"],
             [
               proposedAppInstance.intermediaries![0],
+              // todo(xuanji): doesn't sorting destroy this?
```
Comment (snario) [inline packages/node/src/store.ts]: Pls solve before merging

File: packages/node/src/node.ts [hunk 2]
```diff
@@ -241,25 +241,34 @@ export class Node {
...
-            protocol,
-            transaction
+          verifyFinalCommitment(context.finalCommitment);
+          await this.requestHandler.store.setSetupCommitmentForMultisig(
+            params.multisigAddress,
+            context.finalCommitment!
           );
+        } else {
+          /// install-virtual-app, uninstall-virtual-app
+          console.log("skipping commitment write - not implemented yet");
```
Comment (snario) [inline packages/node/src/node.ts]: Pls make sure there is a story to fix this in the next day or two

File: packages/node/src/node.ts [hunk 3]
```diff
@@ -241,25 +241,34 @@ export class Node {
...
+              "WRITE_COMMITMENT: params did not contain multisigAddress"
             );
           }
-          const transaction = context.commitments[0].transaction([
-            context.signatures[0] // TODO: add counterparty signature
-          ]);
-          if (!context.appIdentityHash) {
-            throw new Error(
-              `appIdentityHash required to store commitment. protocol=${protocol}`
+          if (!getAddress(params.multisigAddress)) {
```
Comment (snario) [inline packages/node/src/node.ts]: Why is this here? Surely it would have been caught well ahead of time in the instruction executor process

File: packages/node/src/node.ts [hunk 4]
```diff
@@ -241,25 +241,34 @@ export class Node {
...
+        } else if (
+          protocol === Protocol.Install ||
+          protocol === Protocol.Uninstall ||
+          protocol === Protocol.Update
+        ) {
+          /// install, update, uninstall,
+          /// All other protocols are stored under the key
+          /// appIdentityHash/protocolName. This means that an app only has one
+          /// update commitment.
+          const params = message.params as any;
```
Comment (snario) [inline packages/node/src/node.ts]: Don't cast to `any`

File: packages/machine/src/protocol/update.ts [hunk 5]
```diff
@@ -85,14 +91,20 @@ export const UPDATE_PROTOCOL: ProtocolExecutionFlow = {
     // Sign the same state update yourself
     Opcode.OP_SIGN,
 
+    (message: ProtocolMessage, context: Context) => {
+      context.finalCommitment = context.commitments[0].transaction([
+        context.signatures[0],
+        message.signature!
+      ]);
```
Comment (snario) [inline packages/machine/src/protocol/update.ts]: For code clarity could we name these values?

e.g.,

```ts
    function generateSignedCommitment(message: ProtocolMessage, context: Context) => {
        const setStateCommitment = context.commitments[0] as SetStateCommitment;
        const mySignature = context.signatures[0];
        const counterpartySignature = message.signature as Signature;
	      context.finalCommitment = setStateCommitment.transaction([
            mySignature, 
            counterpartySignature
        ]);
    }

File: packages/node/src/node.ts (hunk 4 above)
Comment (IIIIllllIIIIllllIIIIllllIIIIllllIIIIll) [reply packages/node/src/node.ts]: Typescript doesn't let me access `params.multisigAddress` without the cast

File: packages/machine/src/protocol/update.ts (hunk 5 above)
Comment (IIIIllllIIIIllllIIIIllllIIIIllllIIIIll) [reply packages/machine/src/protocol/update.ts]: good idea

File: packages/node/src/node.ts (hunk 4 above)
Comment (snario) [reply packages/node/src/node.ts]: Cast it to what it is, not `any`. In this case it would be `SetupParams | InstallParams | UninstallParams`
